### PR TITLE
Now uses Ruby 2.4 (previously 2.2) and builds on Catalina 10.15 (previously El Capitan 10.11)

### PR DESCRIPTION
**Pull request by joshdholtz**
Fixes https://github.com/fastlane/fastlane/issues/15496

### Problem
- Packaged _fastlane_ used Ruby 2.2 which is too old for some _fastlane_ dependencies and causes installation issues leading to packaged _fastlane_ installed version `2.28.3`
- Packaged _fastlane_ could also only be built on OSX 10.1

### Goal
- Build new packaged _fastlane_ with Ruby 2.4 on Catalina

## Description
- Removed `aws-sdk-v1` which had a dependency of `nokogiri` as this was only used for the Fabric app
- Removed the namespace and building of the Fabric app
- Removed using the `RubyTask` to manually build Ruby 2.2
  - Replaced with [ruby-build](https://github.com/rbenv/ruby-build) (which is what [rbenv](https://github.com/rbenv/rbenv) uses) which now builds Ruby 2.4 with OpenSSL
- Removed the building of installed OSX gems (I think this is okay??? 🤔)
  - Seemed to be causing problems and was full of hacks/patches

## Results
Screen Shot 2019-11-05 at 6 04 38 PM

### CODE REVIEW

**Comment by joshdholtz** [comment]
Turns out this currently has some harcoded absolute paths to `/Users/josh` 😱 Need to look into 😔

**Comment by KrauseFx** [comment]
Wow!

File: bundle_tasks.rake [hunk 1]
```diff
@@ -145,32 +145,9 @@ module FastlaneRake
   @@installed_openssl = openssl_tasks.installed_path
   def self.openssl_task
     @@installed_openssl
+    raise "ugh - #{@@installed_openssl}"
```
Comment (snatchev) [inline bundle_tasks.rake]: 💣

File: bundle_tasks.rake [hunk 2]
```diff
@@ -210,7 +187,7 @@ module FastlaneRake
   # ------------------------------------------------------------------------------
   @@installed_bundler_bin = File.join(BUNDLE_DESTROOT, 'bin/bundle')
   file @@installed_bundler_bin => rubygems_update_dir do
-    install_gem 'bundler'
+    install_gem 'bundler', '1.13.6'
```
Comment (snatchev) [inline bundle_tasks.rake]: Okay to lock on this specific version?  Maybe '~> 1.13' is fine?

File: Rakefile [hunk 3]
```diff
@@ -14,7 +13,7 @@ WORKBENCH_DIR = FastlaneRake::WORKBENCH_DIR
 DOWNLOAD_DIR = FastlaneRake::DOWNLOAD_DIR
 DESTROOT = FastlaneRake::DESTROOT
 FASTLANE_GEM_VERSION = FastlaneRake::FASTLANE_GEM_VERSION
-XCODE_PATH = '/Applications/Xcode-7.3.1.app'
+RUBY__VERSION = FastlaneRake::RUBY__VERSION
```
Comment (snatchev) [inline Rakefile]: The double-underscore looks like a mistake. Intentional on `FastlaneRake` ? Also, doesn't extending bring in the constants already?

Comment (snatchev) [reply Rakefile]: Edit, I see the need for the double-under (`RUBY_VERSION` is already defined). Lets call it something else. `INSTALL_VERSION` or something.

File: setup.rb [hunk 4]
```diff
@@ -108,7 +108,7 @@ module FastlaneRake
   READLINE_VERSION = '6.3'
   READLINE_URL = "http://ftpmirror.gnu.org/readline/readline-#{READLINE_VERSION}.tar.gz"
 
-  RUBY__VERSION = '2.2.4'
+  RUBY__VERSION = '2.4.5'
   RUBY_URL = "http://cache.ruby-lang.org/pub/ruby/2.2/ruby-#{RUBY__VERSION}.tar.gz"
```
Comment (snatchev) [inline setup.rb]: Ah, I see need for the double-under.

`http://cache.ruby-lang.org/pub/ruby/2.2/ruby-2.4.5.tar.gz` returns a 404. `RUBY_URL` looks unused.

File: Rakefile [hunk 5]
```diff
@@ -1,10 +1,9 @@
-require 'aws-sdk-v1'
 require 'json'
 
 load './bundle_tasks.rake'
 extend FastlaneRake
 
-BUNDLE_VERSION = 1.8
+BUNDLE_VERSION = 1.9
```
Comment (ohayon) [inline Rakefile]: Feels like this is major version bump worthy?

File: Rakefile [hunk 6]
```diff
@@ -107,16 +74,23 @@ namespace :package do
...
   desc "fastlane-#{FASTLANE_GEM_VERSION}"
   task :version do
     puts "fastlane-#{FASTLANE_GEM_VERSION}"
   end
 
-  task :build_ruby => FastlaneRake.ruby_task
+  task :build_ruby do
+    full_path = File.join(DESTROOT, 'bundle')
+
+    command = "/usr/local/bin/ruby-build #{RUBY__VERSION} #{full_path}"
```
Comment (ohayon) [inline Rakefile]: Doe we require that `ruby-build` is already installed?

File: Rakefile [hunk 7]
```diff
@@ -107,16 +74,23 @@ namespace :package do
...
+    puts "Install Command: #{command}"
+
+    success = system({'RUBY_CONFIGURE_OPTS' => '--with-openssl-dir=/usr/local/opt/openssl'}, command)
```
Comment (ohayon) [inline Rakefile]: Worth getting the error output here to show on failure?

File: setup.rb [hunk 8]
```diff
@@ -108,7 +108,7 @@ module FastlaneRake
   READLINE_VERSION = '6.3'
   READLINE_URL = "http://ftpmirror.gnu.org/readline/readline-#{READLINE_VERSION}.tar.gz"
 
-  RUBY__VERSION = '2.2.4'
+  RUBY__VERSION = '2.4.5'
```
Comment (ohayon) [inline setup.rb]: Any reason not to use an even later version?

**Comment by joshdholtz** [comment]
Closing this in favor of #99

**Pull request closed without merging**